### PR TITLE
backend: fix error handling in iterateRows

### LINKED ISSUE #21: plugin panic
```
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511359608-07:00 level=debug msg="panic: runtime error: invalid memory address or nil pointer dereference"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511539506-07:00 level=debug msg="[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb8239d]"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511573954-07:00 level=debug msg=
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511625079-07:00 level=debug msg="goroutine 126 [running]:"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.51165421-07:00 level=debug msg="github.com/SnellerInc/grafana-datasource/pkg/plugin.deriveSchema({0xc000668000, 
0x4c03e9, 0x526000})"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511684817-07:00 level=debug msg="\t/home/runner/work/grafana-datasource/grafana-datasource/pkg/plugin/query.go:4
08 +0x1bd"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511716256-07:00 level=debug msg="github.com/SnellerInc/grafana-datasource/pkg/plugin.frameFromSnellerResult({0x1
3a81e8, 0x1}, {0xc0003f66e0, 0x97}, {0x7e977378d9d8?, 0xc000091540?}, {0x0, 0x0})"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511755874-07:00 level=debug msg="\t/home/runner/work/grafana-datasource/grafana-datasource/pkg/plugin/query.go:28 +0x8e"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511789953-07:00 level=debug msg="github.com/SnellerInc/grafana-datasource/pkg/plugin.(*Datasource).query(0xc00057a5a0?, {0xe85218, 0xc000499b90}, {0x1, {0xc000285680, 0x1a}, 0xc000425940, 0x0, 0xc0000fac00}, {{0x13a81e8, ...}, ...})"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511820096-07:00 level=debug msg="\t/home/runner/work/grafana-datasource/grafana-datasource/pkg/plugin/datasource.go:324 +0x1446"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511848601-07:00 level=debug msg="github.com/SnellerInc/grafana-datasource/pkg/plugin.(*Datasource).handleQuery.func1({{0x13a81e8, 0x1}, {0x0, 0x0}, 0x850, 0x2540be400, {{0x18054ac0, 0xedc2b5562, 0x13f7040}, {0x18054ac0, ...}}, ...})"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511878771-07:00 level=debug msg="\t/home/runner/work/grafana-datasource/grafana-datasource/pkg/plugin/datasource.go:248 +0x10b"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511907924-07:00 level=debug msg="created by github.com/SnellerInc/grafana-datasource/pkg/plugin.(*Datasource).handleQuery"
logger=plugin.sneller-sneller-datasource t=2023-06-26T08:46:12.511937337-07:00 level=debug msg="\t/home/runner/work/grafana-datasource/grafana-datasource/pkg/plugin/datasource.go:247 +0x2e9"
```

### PR DESCRIPTION
**Pull request by philhofer**
Fixes #21

### CODE REVIEW

File: pkg/plugin/query.go [hunk 1]
```diff
@@ -541,22 +541,23 @@ func iterateRows(buf []byte, readRowFn func(reader *IonReader, index int) error)
...
 		err = reader.StepOut()
 		if err != nil {
-			break
+			return nil, err
 		}
-
 		index++
 	}
-
+	if status == nil {
```
Comment (flobernd) [inline pkg/plugin/query.go:558]: This is already checked on a higher level (might make sense to check it here instead, but in this case you might want to remove the existing check).

**Review by flobernd — APPROVED** [review]
LGTM!

File: pkg/plugin/query.go (hunk 1 above)
Comment (philhofer) [reply pkg/plugin/query.go:558]: The nil check isn't present in `deriveSchema`, which is why there was a panic:
```go
	status, err := iterateRows(buf, func(reader *IonReader, index int) error {
		schema.RowCount += 1
		return analyzeRow(reader, &schema, lookup)
	})
	if err != nil {
		return nil, err
	}
	schema.FinalStatus = status
	// Detect missing values
	for _, col := range schema.Columns {
		if col.Count != schema.RowCount {
			col.Optional = true
		}
	}
	if status.ResultSet.IsEmpty() { // <-- panic if status == nil and err == nil
		return &schema, nil
	}
```

Comment (flobernd) [reply pkg/plugin/query.go:558]: Ahh makes sense! The existing check is on an even higher level.

**Pull request merged**